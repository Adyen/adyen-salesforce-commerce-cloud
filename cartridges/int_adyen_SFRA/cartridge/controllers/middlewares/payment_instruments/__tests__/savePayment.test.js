"use strict";

/* eslint-disable global-require */
var savePayment;
var res;
var req;
beforeEach(function () {
  var _require = require('../../index'),
      paymentInstruments = _require.paymentInstruments;

  savePayment = paymentInstruments.savePayment;
  jest.clearAllMocks();
  res = {
    json: jest.fn()
  };
  req = {
    currentCustomer: {
      profile: {
        customerNo: 'mocked_customerNo'
      }
    }
  };
});
afterEach(function () {
  jest.resetModules();
});
describe('Save Payment', function () {
  it('should do nothing if adyen secured fields is not enabled', function () {
    var AdyenHelper = require('*/cartridge/scripts/util/adyenHelper');

    var server = require('server');

    AdyenHelper.getAdyenSecuredFieldsEnabled.mockImplementation(function () {
      return false;
    });
    savePayment.call({
      emit: jest.fn()
    }, req, res, jest.fn());
    expect(server.forms.getForm).toBeCalledTimes(0);
  });
  it('should fail if zeroAuth has error', function () {
    var adyenZeroAuth = require('*/cartridge/scripts/adyenZeroAuth');

    adyenZeroAuth.zeroAuthPayment.mockImplementation(function () {
      return {
        error: true
      };
    });
    savePayment.call({
      emit: jest.fn()
    }, req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });
  it('should fail if resultCode is not Authorised', function () {
    var adyenZeroAuth = require('*/cartridge/scripts/adyenZeroAuth');

    adyenZeroAuth.zeroAuthPayment.mockImplementation(function () {
      return {
        resultCode: 'Not_Authorised'
      };
    });
    savePayment.call({
      emit: jest.fn()
    }, req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });
  it('should succeed', function () {
    savePayment.call({
      emit: jest.fn()
    }, req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });
});