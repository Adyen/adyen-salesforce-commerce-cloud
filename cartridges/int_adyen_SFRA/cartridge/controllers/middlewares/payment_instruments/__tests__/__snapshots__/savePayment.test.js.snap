// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Save Payment should fail if resultCode is not Authorised 1`] = `
Array [
  Array [
    Object {
      "error": Array [
        "mocked_error.card.information.error",
      ],
      "success": false,
    },
  ],
]
`;

exports[`Save Payment should fail if zeroAuth has error 1`] = `
Array [
  Array [
    Object {
      "error": Array [
        "mocked_error.card.information.error",
      ],
      "success": false,
    },
  ],
]
`;

exports[`Save Payment should succeed 1`] = `
Array [
  Array [
    Object {
      "redirectUrl": "[\\"PaymentInstruments-List\\"]",
      "success": true,
    },
  ],
]
`;
